### PR TITLE
Implement plan intent service

### LINKED ISSUE #218: Implement PlanIntentService
Intent service that fetches from these [endpoints](https://github.com/OpenSRP/opensrp-server-web/pull/47).

### PR DESCRIPTION
**Pull request by vincent-karuri**
Closes #218.

### CODE REVIEW

File: opensrp-app/src/main/java/org/smartregister/AllConstants.java [hunk 1]
```diff
@@ -98,6 +98,7 @@
 
     public static final String CAMPAIGNS = "CAMPAIGNS";
     public static final String OPERATIONAL_AREAS = "OPERATIONAL_AREAS";
+    public static final String CURRENT_OPERATIONAL_AREA = "CURRENT_OPERATIONAL_AREA";
```
Comment (githengi) [inline opensrp-app/src/main/java/org/smartregister/AllConstants.java]: This is not needed Plans should be downloaded for all `OPERATIONAL_AREAS`

File: opensrp-app/src/main/java/org/smartregister/sync/helper/PlanIntentServiceHelper.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        try {
+            long serverVersion = 0;
+            try {
+                serverVersion = Long.parseLong(allSharedPreferences.getPreference(PLAN_LAST_SYNC_DATE));
+            } catch (NumberFormatException e) {
+                Log.e(TAG, e.getMessage(), e);
+            }
+            if (serverVersion > 0) { serverVersion += 1; };
+            // fetch and save plans
+            String plansResponse = fetchPlans(allSharedPreferences.getPreference(CURRENT_OPERATIONAL_AREA), serverVersion);
```
Comment (githengi) [inline opensrp-app/src/main/java/org/smartregister/sync/helper/PlanIntentServiceHelper.java]: fetch plans for all `OPERATIONAL_AREAS`

**Comment by githengi** [comment]
Update the `TaskServiceHelper` to download all tasks for the existing Plans

File: opensrp-app/src/main/java/org/smartregister/AllConstants.java (hunk 1 above)
Comment (vincent-karuri) [reply opensrp-app/src/main/java/org/smartregister/AllConstants.java]: Done.

File: opensrp-app/src/main/java/org/smartregister/sync/helper/PlanIntentServiceHelper.java (hunk 2 above)
Comment (vincent-karuri) [reply opensrp-app/src/main/java/org/smartregister/sync/helper/PlanIntentServiceHelper.java]: Done.

**Pull request merged**